pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -1303,13 +1312,14 @@ public static String createRandom(int len) {
     }
 
     public static Address getToAddressId(@NonNull EasyPostUtil easyPostUtil, KitRequestSettings kitRequestSettings, String addressId,
-                                         DDPParticipant participant, DDPInstanceDto ddpInstanceDto) throws Exception {
+                                         DDPParticipant participant, DDPInstanceDto ddpInstanceDto) throws EasyPostException {
         Address toAddress = null;
-        if (addressId == null && participant == null) { //if both are set to null then it is return label!
-            toAddress =
-                    easyPostUtil.createAddressWithoutValidation(kitRequestSettings.getReturnName(), kitRequestSettings.getReturnStreet1(),
-                            kitRequestSettings.getReturnStreet2(), kitRequestSettings.getReturnCity(), kitRequestSettings.getReturnZip(),
-                            kitRequestSettings.getReturnState(), kitRequestSettings.getReturnCountry(), kitRequestSettings.getPhone());
+        if (isReturnAddress(addressId, participant)) {
+            try {
+                toAddress = createReturnAddress(kitRequestSettings, easyPostUtil);
+            } catch (EasyPostException e) {
+                logger.error("Couldn't generate return address for a kit for ddpInstance " + kitRequestSettings.getDdpInstanceId(), e);
```

Comment:
In this situation, what happens when the callers receive a `null` address?  Is that something that we should look at for improving robustness?

Reply:
Great catch!! Thank you!! I added a check for it before returning `toAddress`